app_flutter/lib/agent_dashboard_page.dart

```diff
@@ -0,0 +1,221 @@
+      create: (_) => agentState,
+      child: AgentDashboard(
+        scaffoldKey: _scaffoldKey,
+        agentState: agentState,
+        agentFilter: agentFilter,
+      ),
+    );
+  }
+
+  void _showErrorSnackbar() {
+    final Row snackbarContent = Row(
+      children: <Widget>[
+        const Icon(Icons.error),
+        const SizedBox(width: 10),
+        Text(agentState.errors.message)
+      ],
+    );
+    _scaffoldKey.currentState.showSnackBar(
+      SnackBar(
+        content: snackbarContent,
+        backgroundColor: Theme.of(context).errorColor,
+        duration: AgentDashboardPage.errorSnackbarDuration,
+      ),
+    );
+  }
+
+  @override
+  void dispose() {
+    agentState.errors.removeListener(_showErrorSnackbar);
+    super.dispose();
+  }
+}
+
+/// Shows current status of Flutter infra agents.
+class AgentDashboard extends StatefulWidget {
+  const AgentDashboard({
+    this.scaffoldKey,
+    this.agentState,
+    this.agentFilter,
+  });
+
+  final GlobalKey<ScaffoldState> scaffoldKey;
+  final AgentState agentState;
+  final String agentFilter;
+
+  @override
+  _AgentDashboardState createState() => _AgentDashboardState();
+}
+
+class _AgentDashboardState extends State<AgentDashboard> {
+  TextEditingController _agentIdController;
+  TextEditingController _agentCapabilitiesController;
+
+  @override
+  void initState() {
+    super.initState();
+
+    _agentIdController = TextEditingController();
+    _agentCapabilitiesController = TextEditingController();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Consumer<AgentState>(
+      builder: (_, AgentState agentState, Widget child) => Scaffold(
+        key: widget.scaffoldKey,
+        appBar: AppBar(
+          title: const Text('Infra Agents'),
+          actions: <Widget>[
+            SignInButton(authService: agentState.authService),
+          ],
+        ),
+        body: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            // Without this container the ListView will not be centered.
+            Container(),
+            Expanded(
+              child: SizedBox(
+                width: 500,
+                child: AgentList(
+                  agents: agentState.agents,
+                  agentState: agentState,
+                  agentFilter: widget.agentFilter,
+                ),
+              ),
+            ),
+          ],
+        ),
+        drawer: const NavigationDrawer(
+          currentRoute: AgentDashboardPage.routeName,
+        ),
+        floatingActionButton: RaisedButton(
+          child: const Text(
+            'Create Agent',
+            textScaleFactor: 1.5,
+          ),
+          color: Theme.of(context).primaryColor,
+          padding: const EdgeInsets.all(10.0),
+          onPressed: () => _showCreateAgentDialog(context, agentState),
+        ),
+      ),
+    );
+  }
+
+  void _showCreateAgentDialog(BuildContext context, AgentState agentState) {
```

Comment:
This is complex enough to be its own widget

Reply:
Thanks for pointing this out! Refactoring this out was able to simplify the rest of the widget into a stateless widget, only making the dialog part a stateful widget.